conn_bench_test.go

```diff
@@ -0,0 +1,47 @@
+package libp2pquic
+
+import (
+	"context"
+	"crypto/rand"
+	"testing"
+
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+func BenchmarkInterceptSecured(b *testing.B) {
```

Comment:
@marten-seemann @Stebalien 

Figured we might as well write a little benchmark to know how much state is being constructed by QUIC after a successful handshake, which we could prevent by blocking denied connections earlier.

TL;DR: we're looking at a 20% difference in allocs (1000+ allocs), and 10% difference in bytes (50kB per op).

## This PR

```
goos: darwin
goarch: amd64
pkg: github.com/libp2p/go-libp2p-quic-transport
BenchmarkInterceptSecured
BenchmarkInterceptSecured-8          694       1673938 ns/op      497751 B/op       4394 allocs/op
PASS

Process finished with exit code 0
```

## PR #157 

```
goos: darwin
goarch: amd64
pkg: github.com/libp2p/go-libp2p-quic-transport
BenchmarkInterceptSecured
BenchmarkInterceptSecured-8          670       1760618 ns/op      544251 B/op       5436 allocs/op
PASS

Process finished with exit code 0
```